packages/client-sdk-nodejs/src/internal/grpc/client-timeout-interceptor.ts

```diff
@@ -1,14 +1,123 @@
-import {InterceptingCall, Interceptor} from '@grpc/grpc-js';
+import {MomentoLogger, MomentoLoggerFactory} from '@gomomento/sdk-core';
+import {InterceptingCall, Interceptor, status} from '@grpc/grpc-js';
+import {RetryStrategy} from '../../config/retry/retry-strategy';
+import {DefaultStorageRetryStrategy} from '../../config/retry/storage-default-retry-strategy';
+import {DefaultMomentoLoggerFactory} from '../../config/logging/default-momento-logger';
 
+// Determine which retry strategy is specified in the configuration
+// and which interceptor to use.
 export const ClientTimeoutInterceptor = (
-  requestTimeoutMs: number
+  overallRequestTimeoutMs: number,
+  retryStrategy?: RetryStrategy,
+  loggerFactory?: MomentoLoggerFactory
 ): Interceptor => {
-  return (options, nextCall) => {
-    if (!options.deadline) {
-      const deadline = new Date(Date.now());
-      deadline.setMilliseconds(deadline.getMilliseconds() + requestTimeoutMs);
-      options.deadline = deadline;
-    }
-    return new InterceptingCall(nextCall(options));
-  };
+  if (
+    retryStrategy !== undefined &&
+    retryStrategy instanceof DefaultStorageRetryStrategy
```

Comment:
I think I like how you broke the timeout interceptors up into two cases here, but we don't want to do this instanceof check to choose between them. One design goal here is that a user could write their own implementation of `RetryStrategy` and leverage the building blocks we are putting together, and this would preclude that.

I suspect the best way to handle this is going to be to extend the `interface` for `RetryStrategy` to include the (optional) `responseDataReceivedTimeoutMillis` property. Then you can just check that here to decide which timeout strategy to use, and if users write their own retry strategy that adheres to the interface, they will get the behavior too.